src/vivarium/framework/lookup.py

```diff
@@ -1,68 +1,71 @@
 """A set of tools for managing data lookups."""
 from numbers import Number
 from datetime import datetime, timedelta
-import warnings
+from typing import Union, List, Tuple, Callable, TypeVar
 
 import pandas as pd
 
 from vivarium.interpolation import Interpolation
+from vivarium.framework.population import PopulationView
 
+value_type = TypeVar('value_type', Number, timedelta, datetime)
```

Comment:
I haven't seen TypeVar!  I've been using `Union` to construct types.  This seems better!  

I recommend giving it a CamelCase name like `ScalarValue`.  Lowercase stuff used in typing usually refer to python standard types or to numpy dtypes (meaning they likely are thin wrappers around C types), at least from what I've seen.  

`datetime` and `timedelta` being immediate exceptions, of course.  I think those just got named before conventions were standardized and are impossible to change because dealing with time is the worst.